imageio/plugins/pyav.py

```diff
@@ -0,0 +1,839 @@
+            filename = getattr(file_handle, "name", None)
+            extension = self.request.extension or self.request.format_hint
+            if extension is None:
+                raise InitializationError("Can't determine output container to use.")
+
+            # hacky, but beats running our own format selection logic
+            # (since av_guess_format is not exposed)
+            try:
+                setattr(file_handle, "name", filename or "tmp" + extension)
+            except AttributeError:
+                pass  # read-only, nothing we can do
+
+            try:
+                self._container = av.open(file_handle, mode="w", format=container)
+            except av.AVError:
+                resource = (
+                    "<bytes>"
+                    if isinstance(self.request.raw_uri, bytes)
+                    else self.request.raw_uri
+                )
+                raise InitializationError(f"PyAV can not write to `{resource}`")
+
+    def read(
+        self,
+        *,
+        index: int = 0,
+        format: str = None,
+        filter_sequence: List[Tuple[str, Union[str, dict]]] = None,
+        filter_graph: Tuple[dict, List] = None,
+        constant_framerate: bool = None,
+        thread_count: int = 0,
+        thread_type: str = None,
+    ) -> np.ndarray:
+        """Read frames from the video.
+
+        If ``index`` is numerical, this function reads the index-th frame from
+        the file. If ``index`` is None, this function reads all frames from the
+        video, stacks them along the first dimension, and returns a batch of
+        frames.
+
+        Parameters
+        ----------
+        index : int
+            The index of the frame to read, e.g. ``index=5`` reads the 5th
+            frame. If ``None``, read all the frames in the video and stack them
+            along a new, prepended, batch dimension.
+        format : str
+            If not None, convert the data into the given format before returning
+            it. If None (default) return the data in the encoded format if it
+            can be expressed as a strided array; otherwise raise an Exception.
+        filter_sequence : List[str, str, dict]
+            If not None, apply the given sequence of FFmpeg filters to each
+            ndimage. Check the (module-level) plugin docs for details and
+            examples.
+        filter_graph : (dict, List)
+            If not None, apply the given graph of FFmpeg filters to each
+            ndimage. The graph is given as a tuple of two dicts. The first dict
+            contains a (named) set of nodes, and the second dict contains a set
+            of edges between nodes of the previous dict.Check the (module-level)
+            plugin docs for details and examples.
+        constant_framerate : bool
+            If True assume the video's framerate is constant. This allows for
+            faster seeking in side the file. If False, the video is reset before
```

Comment:
```suggestion
            faster seeking inside the file. If False, the video is reset before
```

Reply:
Since this can also be auto-detected, what is the need for making this settable with an arg?

Reply:
The auto-detection checks for the container's general ability to store variable framerate. This doesn't say anything about the actual timings of frames inside a file so we allow the user to control this if needed (because it is _really_ slow to seek inside of files with non-constant framerate).